stride/problem/data.py

```diff
@@ -1500,6 +1501,19 @@ def plot_one(self, id, **kwargs):
 
         return axis
 
+    def _resample(self, factor, new_num, **kwargs):
+        sr_orig = 1
+        sr_new = 1*factor
+        data = resampy.resample(self.data, sr_orig, sr_new, axis=1)  # resample
+
+        interp_num = data.shape[1]
+        if new_num is not None and new_num > interp_num:  # do we need to pad the wavelet array?
+            data = np.pad(data, ((0, 0), (0, new_num-interp_num)), axis=1)
+        elif new_num is not None and new_num < interp_num:  # do we need to crop the wavelet array?
+            data = data[:, :new_num]
+
+        return data
```

Comment:
Resampling of a `Traces` objects should not be an in-place operation, because the number of timesteps might have changed. The method should create a new `Traces` object, with the updated time grid. This object will already had the right sampling and length in its data, and we only need to assign the resampled array into it.